tests/cpu/AVX512_tests.cpp

```diff
@@ -0,0 +1,514 @@
+        g = _mm512_mul_ps(g, s);
+        b = _mm512_mul_ps(b, s);
+        a = _mm512_mul_ps(a, s);
+        OCIO::AVX512RGBAPack<outBD>::Store(&outImage[i], r, g, b, a);
+    }
+    for (unsigned i = 0; i < outImage.size(); i++)
+    {
+        float v = (float)inImage[i] * scale;
+
+        if (OCIO::BitDepthInfo<outBD>::isFloat)
+            v = (OutType)v; // casts to half if format is half
+        else
+            v = rintf(v);
+
+        OCIO_CHECK_ASSERT_MESSAGE(!OCIO::FloatsDiffer(v, (float)outImage[i], 0, false),
+                                  GetErrorMessage(v, (float)outImage[i], inBD, outBD));
+    }
+
+    // Test Load/Store Masked
+    for (unsigned pixel_count = 0; pixel_count <= 16; pixel_count++)
+    {
+        __m512 r, g, b, a;
+        // reset all values to zero
+        for (unsigned i = 0; i < outImage.size(); i++)
+        {
+            outImage[i] = 0;
+        }
+
+        OCIO::AVX512RGBAPack<inBD>::LoadMasked(&inImage[0], r, g, b, a, pixel_count);
+        r = _mm512_mul_ps(r, s);
+        g = _mm512_mul_ps(g, s);
+        b = _mm512_mul_ps(b, s);
+        a = _mm512_mul_ps(a, s);
+        OCIO::AVX512RGBAPack<outBD>::StoreMasked(&outImage[0], r, g, b, a, pixel_count);
+
+        for (unsigned i = 0; i < outImage.size(); i++)
+        {
+            float v = (float)inImage[i] * scale;
+
+            // values geater then the pixel count should not have been written to
+            if (i >= pixel_count*4)
+                v = 0.0f;
+
+            if (OCIO::BitDepthInfo<outBD>::isFloat)
+                v = (OutType)v; // casts to half if format is half
+            else
+                v = rintf(v);
+
+            OCIO_CHECK_ASSERT_MESSAGE(!OCIO::FloatsDiffer(v, (float)outImage[i], 0, false),
+                                       GetErrorMessage(v, (float)outImage[i], inBD, outBD));
+        }
+    }
+}
+
+template<OCIO::BitDepth inBD>
+void testConvert_InBitDepth(OCIO::BitDepth outBD)
+{
+    switch(outBD)
+    {
+        case OCIO::BIT_DEPTH_UINT8:
+            return testConvert_OutBitDepth<inBD, OCIO::BIT_DEPTH_UINT8>();
+        case OCIO::BIT_DEPTH_UINT10:
+            return testConvert_OutBitDepth<inBD, OCIO::BIT_DEPTH_UINT10>();
+        case OCIO::BIT_DEPTH_UINT12:
+            return testConvert_OutBitDepth<inBD, OCIO::BIT_DEPTH_UINT12>();
+        case OCIO::BIT_DEPTH_UINT16:
+            return testConvert_OutBitDepth<inBD, OCIO::BIT_DEPTH_UINT16>();
+        case OCIO::BIT_DEPTH_F16:
+            return testConvert_OutBitDepth<inBD, OCIO::BIT_DEPTH_F16>();
+            break;
+        case OCIO::BIT_DEPTH_F32:
+            return testConvert_OutBitDepth<inBD, OCIO::BIT_DEPTH_F32>();
+
+        case OCIO::BIT_DEPTH_UINT14:
+        case OCIO::BIT_DEPTH_UINT32:
+        case OCIO::BIT_DEPTH_UNKNOWN:
+        default:
+            break;
+    }
+}
+
+}
+
+DEFINE_SIMD_TEST(packed_uint8_to_float_test)
+{
+    std::vector<uint8_t> inImage(256);
+    std::vector<float> outImage(256);
```

Comment:
256 --> 512?

Reply:
This test is correct. The test is to check all 8 bit values, translate to float as expected.

Reply:
Is that a case where using similar code compared to subsequent tests would help make it clearer?
```
size_t maxValue = OCIO::BitDepthInfo<OCIO::BIT_DEPTH_UINT8>::maxValue + 1;
std::vector<uint8_t> inImage(maxValue);
std::vector<float> outImage(maxValue);
```

Reply:
@remia I think that is clearer. I can add that to the PR.